src/services/distributions.ts

```diff
@@ -51,25 +51,25 @@ export class DistributionsService {
   async query(options: QueryOptions): Promise<QueryResult> {
     const args = Joi.attempt(options, schemaQuery);
     this.logger.info(`Preparing to query source "${args.source}"...`);
-    const dataset = await this.catalog.getByIdentifier(args.source);
+    const dataset = await this.catalog.getDatasetByDistributionIri(args.source);
     if (dataset === undefined) {
       throw Error(
-        `Source with identifier "${args.source}" not found in catalog`
+        `Source with distribution IRI "${args.source}" not found in catalog`
```

Comment:
This error may bubble up to a user (e.g. via the CLI). Perhaps avoid the word 'distribution' (and 'catalog' and 'IRI'?) and merely use `Source with URI xxx not found` (words that our interfaces [CLI, GraphQL] use)?